### DISCUSSION TITLE
Shapely - How to make the base value reflect the hack cases?

### DISCUSSION THREAD

**Discussion by vallipichappan** [shap/shap#1485] [General]
I have a dataset that has 20% of the data labelled as 1 (i.e. hack) and 80%  0 (i.e. non-hack). I have used the random forest model for the training and prediction of my dataset. While using shapely's force plot, the base value seems to be 0.8.  How do I make the base value start from 0.2 instead (because only 20% of my data is labelled as 1)?

Currently, I'm training my model using inverted y_train and y_test (i.e. Hack = 0 and Non-Hack = 1). I would like to avoid doing this and directly use the force_plot method (and its attributes) to set the base value as 0.2 instead of 0.8. Thanks!

**Comment by vallipichappan** [comment]
https://stackoverflow.com/questions/64201623/shapely-change-the-calculation-of-base-value

Please help!!

**Reply by JyotiPrakashBal** [reply]
shap.force_plot(
        0.2, # set base value to 0.2
        np.hstack([shap_values[0, :], explainer.expected_value]), # add expected value as a bar in the force plot
        pd.concat(
            [c_features, pd.DataFrame({"Base value": [explainer.expected_value]})],
            axis=1,
        ).iloc[0, :], # grabs the identities and values of components
        text_rotation=30, # easier to read
        matplotlib=True,
        show=False, # allows saving, etc.
    )
